src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/Utils.groovy

```diff
@@ -179,5 +206,45 @@ public class Utils {
         r.addAction(new ExecutionModelAction(stages))
     }
 
+    /**
+     * Creates and sets the loading for a cache of {@link Describable}s descending from the given descriptor type.
+     *
+     * @param type The {@link Descriptor} class whose extensions we want to find.
+     * @param excludedSymbols Optional list of symbol names to exclude from the cache.
+     * @return A {@link LoadingCache} for looking up types from symbols.
+     */
+    static generateTypeCache(Class<? extends Descriptor> type, List<String> excludedSymbols = []) {
+        return CacheBuilder.newBuilder()
+            .expireAfterWrite(10, TimeUnit.MINUTES)
+            .build(new CacheLoader<Object, Map<String, String>>() {
+            @Override
+            Map<String, String> load(Object key) throws Exception {
+                return populateTypeCache(type, excludedSymbols)
```

Comment:
There were Reasons. =) I can't remember the exact reasoning at the time, but the idea was to take advantage of thread safety, periodic refreshes, etc with `LoadingCache`, but load everything into it at once rather than re-querying `ExtensionList` every time.